### PR TITLE
pkg/apiserver: split up monolithic package

### PR DESCRIPTION
**Pull request by sttts**
**Based on** https://github.com/kubernetes/kubernetes/pull/38191

This is a first step to integrate pkg/apiserver with pkg/genericapiserver into a common package structure. For this
- pkg/apiserver is cleaned up from code which does not belong there, 
- split up into pkg/apiserver, pkg/apiserver/handlers{,/negotation,/helpers,/errors}

### CODE REVIEW

File: cmd/kubernetes-discovery/pkg/apiserver/handler_apis.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"k8s.io/kubernetes/pkg/api"
 	apierrors "k8s.io/kubernetes/pkg/api/errors"
 	metav1 "k8s.io/kubernetes/pkg/apis/meta/v1"
-	"k8s.io/kubernetes/pkg/apiserver"
+	apiserverhelpers "k8s.io/kubernetes/pkg/apiserver/handlers/helpers"
```
Comment (deads2k) [inline cmd/kubernetes-discovery/pkg/apiserver/handler_apis.go]: another "stuff" package?

File: pkg/genericapiserver/mux/container.go [hunk 2]
```diff
@@ -48,9 +55,42 @@ func NewAPIContainer(mux *http.ServeMux, s runtime.NegotiatedSerializer) *APICon
 	}
 	c.Container.ServeMux = mux
 	c.Container.Router(restful.CurlyRouter{}) // e.g. for proxy/{kind}/{name}/{*}
-
-	apiserver.InstallRecoverHandler(s, c.Container)
-	apiserver.InstallServiceErrorHandler(s, c.Container)
+	c.Container.RecoverHandler(func(panicReason interface{}, httpWriter http.ResponseWriter) {
```
Comment (deads2k) [inline pkg/genericapiserver/mux/container.go]: I dislike inline functions here if we can avoid it.

**Comment by deads2k** [comment]
helpers needs a better name and it needs a squash, but as an intermediate step it's ok.

**Comment by deads2k** [comment]
lgtm

**Comment by sttts** [comment]
Renamed helpers to responsewriters.

**Comment by sttts** [comment]
@k8s-bot bazel test this

**Pull request merged**